src/net_processing.cpp

```diff
@@ -2967,8 +3072,68 @@ static bool SendRejectsAndCheckIfBanned(CNode* pnode, CConnman* connman, bool en
     return false;
 }
 
+/** Check if any dandelion txs expired. If so, relay them. */
+static void CheckDandelionExpiry(CNode* from, CTxMemPool& tx_pool, CConnman* connman)
+{
+    if (from->m_cache_expiry.empty()) return;
+
+    const int64_t now_millis = GetTimeMillis();
+
+    // Constant time lookup:
+    if (from->m_cache_expiry.top() >= now_millis) return;
+
+    // Otherwise walk the whole cache to look for the expired tx(s)
+    do {
+        from->m_cache_expiry.pop();
+    } while (from->m_cache_expiry.top() < now_millis);
+
+    for (auto it = from->m_cache_dandelion.cbegin(); it != from->m_cache_dandelion.cend(); ++it) {
```

Comment:
An idea: if you define `m_cache_expiry` as `std::map<int64_t, CTransactionRef>` - you will get the same 'Constant time lookup' but you won't need to walk the whole cache - expired transactions are already available